python/src/QuantileConfidence_doc.i.in

```diff
@@ -0,0 +1,436 @@
+
+It is given by:
+
+.. math::
+
+    k_{low} = \overline{F}^{-1}_{n, \alpha}(\beta)
+
+where :math:`\overline{F}^{-1}_{n, \alpha}` is the complementary quantile function 
+of the binomial distribution with parameters :math:`n` and :math:`\alpha`.
+The problem has a solution only if:
+
+.. math::
+
+    1 - (1 - \alpha)^n \geq \beta.
+
+The upper tail rank is the smallest integer :math:`k_{up}` such that:
+
+.. math::
+
+    \Pset(x_{\alpha} \leq X_{(k)}) \geq \beta.
+
+The solution is:
+
+.. math::
+
+    k_{up} = F^{-1}_{n, \alpha}(\beta)
+
+where :math:`F^{-1}_{n, \alpha}` is the quantile function of the cumulative distribution function of the 
+binomial distribution with parameters :math:`n` and :math:`\alpha`.
+The problem has a solution only if:
+
+.. math::
+
+    1 - \alpha^n \geq \beta.
+
+Parameters
+----------
+size : int
+    Sample size
+tail : bool, optional
+    Whether to return the lower or upper tail rank
+
+Returns
+-------
+rank : int
+    Rank :math:`k \in \{0, \hdots, n-1\}`
+
+Examples
+--------
+>>> import openturns as ot
+>>> import openturns.experimental as otexp
+>>> alpha = 0.05
+>>> beta = 0.95
+>>> algo = otexp.QuantileConfidence(alpha, beta)
+>>> rank = algo.computeUnilateralRank(100)
+"
+
+// ---------------------------------------------------------------------
+
+%feature("docstring") OT::QuantileConfidence::computeBilateralRank
+"Evaluate the bilateral rank of a quantile.
+
+The bilateral ranks :math:`k_1, k_2` are the integers minimizing the probability:
+
+.. math::
+
+    \argmin_{0 \leq k_1 \leq k_2 \leq n - 1 \, | \, \Pset(k_1 < X \leq k_2) \geq \beta} \Pset(k_1 < X \leq k_2) | 
+
+where :math:`X \sim \cB(n, \alpha)`.
+These ranks exist only if:
+
+.. math::
+
+    1 - \alpha^n - (1-\alpha)^n \ge \beta.
+
+Parameters
+----------
+size : int
+    Sample size :math:`n`
+
+Returns
+-------
+ranks : :class:`~openturns.Indices` of size 2
+    Pair of lower and upper ranks :math:`(k_1, k_2)` such that :math:`0 \leq k_1 \leq k_2 \leq n - 1`.
+
+Examples
+--------
+>>> import openturns as ot
+>>> import openturns.experimental as otexp
+>>> alpha = 0.05
+>>> beta = 0.95
+>>> algo = otexp.QuantileConfidence(alpha, beta)
+>>> k1, k2 = algo.computeBilateralRank(100)
+"
+
+// ---------------------------------------------------------------------
+
+%feature("docstring") OT::QuantileConfidence::computeUnilateralConfidenceInterval
+"Evaluate an unilateral confidence interval of a quantile.
+
+The lower tail confidence interval is given by the order statistics:
+
+.. math::
+
+    [X_{(k_{low})}, +\infty]
+
+and the upper tail confidence interval is given by the order statistics:
+
+.. math::
+
+    [-\infty, X_{(k_{up})}]
+
+where the lower or upper ranks :math:`k_{low}, k_{up}` are given by :meth:`computeUnilateralRank`
+
+Parameters
+----------
+sample : 2-d sequence of float
+    Quantile level
+tail : bool, optional
+    Whether to return the lower or upper tail confidence interval. Default is False.
```

Comment:
I do not understand. If True, then it returns the lower or upper tail?

Reply:
indeed, maybe its clearer to write "Whether to return the upper or lower tail confidence interval"